### ISSUE TITLE
Genie_python: cached pvs can return incorrect answers

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#3195]
As a developer, I would like genie_python to return the correct answer when I ask if a PV exists.

It appears that the change in #1188 has introduced some unintended side effects. When asking whether a PV exists, if a PV is in genie_python's cache, it always returns True even if the PV has now disappeared.

- Figure out if this will also be a problem in the blockserver
- Fix IOC system tests
- Fix system tests

**Comment by Tom-Willemsen** [comment]
We think it is an issue in the interface between C and Python in CaChannel. @FreddieAkeroyd has some ideas regarding pre-emptive callbacks in CaChannel.

**Comment by FreddieAkeroyd** [comment]
Having communicated with the CaChannel author Xiaoqiang Wang he suggested that the issue was caused by the handlers causing an incorrect CA context to be created. CaChannel requests a pre-emptive context, but it delays context creation until the first CA call. So if the handlers are installed after a CaChannel object is created but before the first CA call then a default non-preemptive context is created within the CA library which leads to the observed behaviour. I have tested and merged a fix.
